pkgs/development/beam-modules/build-mix.nix

```diff
@@ -0,0 +1,81 @@
+{ stdenv, writeText, elixir, erlang, hex, lib }:
+
+{ name
+, version
+, src
+, buildInputs ? []
+, beamDeps ? []
+, postPatch ? ""
+, compilePorts ? false
+, meta ? {}
+, enableDebugInfo ? false
+, mixEnv ? "prod"
+, ... }@attrs:
+
+with lib;
+
+let
+
+  shell = drv: stdenv.mkDerivation {
+          name = "interactive-shell-${drv.name}";
+          buildInputs = [ drv ];
+    };
+
+  bootstrapper = ./mix-bootstrap;
+
+  pkg = self: stdenv.mkDerivation ( attrs // {
+    name = "${name}-${version}";
+    inherit version;
+
+    dontStrip = true;
+
+    inherit src;
+
+    MIX_ENV = mixEnv;
+    MIX_DEBUG = if enableDebugInfo then 1 else 0;
+    HEX_OFFLINE = 1;
+
+    setupHook = attrs.setupHook or
+      writeText "setupHook.sh" ''
+        addToSearchPath ERL_LIBS "$1/lib/erlang/lib"
+      '';
+
+    inherit buildInputs;
+    propagatedBuildInputs = [ hex elixir ] ++ beamDeps;
```

Comment:
We should definitely remove this `[ hex elixir ] ++` part. I don't think we want symlinks to hex and elixir everywhere.

Reply:
I want to propose the change here
```
buildInputs = buildInputs ++ [ hex elixir ]
propagatedBuildInputs = beamdeps;
```

regardless of the symlinks, I don't think we want to have elixir and hex in propagated build inputs.